ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -828,13 +831,14 @@ func (r *RayClusterReconciler) reconcilePods(ctx context.Context, instance *rayv
 // (1) shouldDelete: Whether the Pod should be deleted.
 // (2) reason: The reason why the Pod should or should not be deleted.
 func shouldDeletePod(pod corev1.Pod, nodeType rayv1.RayNodeType) (bool, string) {
-	// If a Pod's restart policy is set to `Always`, KubeRay will not delete
+	// If a Pod's restart policy is set to `Always` and Pod not be evicted, KubeRay will not delete
 	// the Pod and rely on the Pod's restart policy to restart the Pod.
 	isRestartPolicyAlways := pod.Spec.RestartPolicy == corev1.RestartPolicyAlways
+	isPodEvicted := pod.Status.Reason == podEvictedReason
 
 	// If the Pod's status is `Failed` or `Succeeded`, the Pod will not restart and we can safely delete it.
 	if pod.Status.Phase == corev1.PodFailed || pod.Status.Phase == corev1.PodSucceeded {
-		if isRestartPolicyAlways {
+		if isRestartPolicyAlways && !isPodEvicted {
```

Comment:
It seems like think this if statement down to line 850 can be removed, unless we are know of other reasons a pod phase would be `Failed` where the pod shouldn't be recreated. The comments below suggest we only added this check because we couldn't reproduce the Failed phase when restart policy is Always.

Reply:
(if we keep this if statement, update the comments below on why evicted is treated differently)

Reply:
> It seems like think this if statement down to line 850 can be removed, unless we are know of other reasons a pod phase would be `Failed` where the pod shouldn't be recreated. The comments below suggest we only added this check because we couldn't reproduce the Failed phase when restart policy is Always.

There are errors in this comments, and there was some discussion about this in the [issue](https://github.com/ray-project/kuberay/issues/2125). I will update them then for review

Reply:
code comments updated, please review. @andrewsykim